libtelemetry/src/main/java/com/mapbox/android/telemetry/MetricUtils.java

```diff
@@ -0,0 +1,134 @@
+package com.mapbox.android.telemetry;
+
+import android.content.Context;
+import android.location.Location;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.gson.Gson;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+
+class MetricUtils {
```

Comment:
i'm a bit puzzled about the reasoning behind defining date as an instance variable and the rest being static variables. what is the purpose/responsibility of this class? if it is an utility shouldn't it be closed for instantiation? Now, classic utility class will have all static methods. This one does have a mix of both.